packages/hardhat-ethers/src/types/index.ts

```diff
@@ -11,6 +11,9 @@ export interface FactoryOptions {
   signer?: ethers.Signer;
   libraries?: Libraries;
 }
+export interface DeployOptions extends FactoryOptions {
+  args?: any[];
```

Comment:
The team is leaning towards requiring that contract args be typed in order to accept this PR. Perhaps the interface could be changed to support something like `ethers.deployContract("Foo", [arg1, arg2], options)`?

Reply:
Yep, I'm planning to add overloads to typechain as well, in order args to be typed.

So it could generate overloads similar to [this](https://github.com/dethcrypto/TypeChain/blob/master/packages/hardhat/test/fixture-projects/hardhat-project/typechain-types/hardhat.d.ts). Do you think it can work?

```ts
// ArgsType would be generated by Typechain
// following is an example of an overload for a ContractName.
function deployContract(name: 'ContractName', options: DeployOptions<ArgsType>);

interface DeployOptions<T extends Array> extends FactoryOptions {
  args?: T;
}
```

Reply:
Or do you prefer to handle this in this way:
```ts
ethers.deployContract("Foo", [arg1, arg2], options) // for constructor args
ethers.deployContract("Foo", options) // for no constructor args
```

I think doing this change can make deployContract one-liner, at least for short `args`. So this can be great, thanks.

Reply:
Sorry for the delay. I've made changes to the signature as per your suggestion @feuGeneA. Please let me know or suggest changes. 

Though I didn't understand "requiring that contract args be typed", how do you think args can be typed (better than `any[]`) within the hardhat or what's the typability difference you're seeing in previous approach vs this one? (for me these both approaches seem to be typed).